service/controller/resource/ipam/azure_config_persister.go

```diff
@@ -0,0 +1,53 @@
+package ipam
+
+import (
+	"context"
+	"net"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AzureConfigPersisterConfig struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+type AzureConfigPersister struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func NewAzureConfigPersister(config AzureConfigPersisterConfig) (*AzureConfigPersister, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	p := &AzureConfigPersister{
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+	}
+
+	return p, nil
+}
+
+func (p *AzureConfigPersister) Persist(ctx context.Context, subnet net.IPNet, namespace string, name string) error {
+	cr, err := p.g8sClient.ProviderV1alpha1().AzureConfigs(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cr.Spec.Azure.VirtualNetwork.CIDR = subnet.String()
```

Comment:
In addition to this, I think you need to split the VNET CIDR into POD network and further into {master,worker} subnets.

i.e. Let `subnet` be `10.1.0.0/23`. Split that into half and you get:
- `10.1.0.0/24` for PODs (i.e. `calicoSubnet`)
- `10.1.1.0/24` for nodes.
  - Now split this this into two halves for masters and workers:
    - `10.1.1.0/25` for masters
    - `10.1.1.128/25` for workers

...and store them in CR correspondingly. You can see the current implementation here: https://github.com/giantswarm/kubernetesd/blob/master/service/creator/azure.go#L99-L137

Note that `s.IPAM.CreateSubnet()` performs the allocation and persisting in `crdstorage` - we don't need that.